alacritty_terminal/src/ansi.rs

```diff
@@ -12,6 +13,18 @@ use alacritty_config_derive::ConfigDeserialize;
 use crate::index::{Column, Line};
 use crate::term::color::Rgb;
 
+/// Maximum time before a synchronized update is aborted.
+const SYNC_UPDATE_TIMEOUT: Duration = Duration::from_millis(150);
+
+/// Maximum number of bytes read in one synchronized update (2MiB).
+const SYNC_BUFFER_SIZE: usize = 0x20_0000;
```

Comment:
Kitty has a 1MiB limit iirc. So we double that.

While your strategy is interesting, I determined this using a little bit of a different approach. I checked how much memory it would take to fill every single cell of an overly dense grid (tiny font size on 1440p monitor) with truecolor background, foreground and a character.

That came out at slightly below ~2MiB. I think this makes sense since the goal of a synchronized update would be to update a large part of the screen. With 2MiB it should hopefully be possible to draw whatever you want for any reasonable terminal size without getting interrupted by exceeding the buffer size.